### PR TITLE
Migrate logging_controller.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

**Comment by polina-c** [comment]
blocked by https://github.com/flutter/devtools/pull/3792

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 1]
```diff
@@ -291,60 +286,60 @@ class LoggingController extends DisposableController
...
         e.timestamp,
         summary: '${changedInfo.extension}: ${changedInfo.value}',
       ));
     } else if (e.extensionKind == 'Flutter.Error') {
       // TODO(pq): add tests for error extension handling once framework changes
       // are landed.
-      final RemoteDiagnosticsNode node =
-          RemoteDiagnosticsNode(e.extensionData.data, objectGroup, false, null);
+      final RemoteDiagnosticsNode node = RemoteDiagnosticsNode(
+          e.extensionData!.data, objectGroup, false, null);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/logging_controller.dart:342]: nit: trailing comma

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 2]
```diff
@@ -426,32 +421,32 @@ class LoggingController extends DisposableController
...
             );
 
             if (toStringResult is ErrorRef) {
-              final String errorString = _valueAsString(error);
+              final String? errorString = _valueAsString(error);
               result += '\n\n$errorString';
             } else if (toStringResult is InstanceRef) {
               final String str = await _retrieveFullStringValue(
-                  service, e.isolate, toStringResult);
+                  service, e.isolate!, toStringResult);
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/logging_controller.dart:449]: nit: trailing comma

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 3]
```diff
@@ -503,20 +498,20 @@ class LoggingController extends DisposableController
...
-      for (RemoteDiagnosticsNode property in node.inlineProperties) {
+      for (RemoteDiagnosticsNode property in node.inlineProperties!) {
         summary = _findFirstSummary(property);
         if (summary != null) return summary;
       }
     }
     if (node.childrenNow != null) {
-      for (RemoteDiagnosticsNode child in node.childrenNow) {
-        summary = _findFirstSummary(child);
+      for (RemoteDiagnosticsNode? child in node.childrenNow!) {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/logging/logging_controller.dart:513]: can child be non-nullable

File: packages/devtools_app/test/timeline_streams_test.dart [hunk 4]
```diff
@@ -17,6 +18,8 @@ void main() async {
     const FlutterRunConfiguration(withDebugger: true),
   );
 
+  setGlobal(MessageBus, MessageBus());
```
Comment (kenzieschmoll) [inline packages/devtools_app/test/timeline_streams_test.dart:21]: this shouldn't be necessary now that the global is set in FlutterTestEnvironment

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 5]
```diff
@@ -37,39 +37,41 @@ final DateFormat timeFormat = DateFormat('HH:mm:ss.SSS');
...
-        onUnavailable: (truncatedValue) => fallback,
-      ) ??
-      fallback;
+  return service
+          ?.retrieveFullStringValue(
+            isolateRef.id!,
+            stringRef,
+            onUnavailable: (truncatedValue) => fallback,
+          )
+          .then((value) => value!) ??
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:61]: fixed

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 6]
```diff
@@ -37,39 +37,41 @@ final DateFormat timeFormat = DateFormat('HH:mm:ss.SSS');
...
-    @required this.createLoggingTree,
+    required this.onShowInspector,
+    required this.onShowDetails,
+    required this.createLoggingTree,
   });
 
   static const JsonEncoder jsonEncoder = JsonEncoder.withIndent('  ');
 
-  LogData data;
+  LogData? data;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:74]: done

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 7]
```diff
@@ -81,48 +83,39 @@ class LoggingDetailsController {
...
-          if (node != null) {
-            tree.maybePopulateChildren(node);
-          }
+          final InspectorTreeNode node = tree!.selection!;
+          tree!.maybePopulateChildren(node);
 
           // TODO(jacobr): node.diagnostic.isDiagnosticableValue isn't quite
           // right.
-          if (node.diagnostic.isDiagnosticableValue) {
+          if (node.diagnostic.isDiagnosticableValue!) {
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:101]: It is not nullable, but it is not migrated yet.

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 8]
```diff
@@ -210,14 +204,15 @@ class LoggingController extends DisposableController
...
+      final List<LogData> logs = filteredData.value;
       if (!logs.contains(selected)) {
         _selectedLog.value = null;
       }
     }
   }
 
-  ObjectGroup get objectGroup => serviceManager.consoleService.objectGroup;
+  ObjectGroup? get objectGroup =>
+      serviceManager.consoleService.objectGroup as ObjectGroup?;
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:215]: done

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart [hunk 9]
```diff
@@ -248,12 +243,12 @@ class LoggingController extends DisposableController
     serviceManager.errorBadgeManager.clearErrors(LoggingScreen.id);
   }
 
-  void _handleConnectionStart(VmServiceWrapper service) async {
+  void _handleConnectionStart(VmServiceWrapper? service) async {
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:246]: done

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:342]: done

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:449]: done

File: packages/devtools_app/lib/src/screens/logging/logging_controller.dart (hunk 3 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/logging/logging_controller.dart:513]: done

File: packages/devtools_app/test/timeline_streams_test.dart (hunk 4 above)
Comment (polina-c) [reply packages/devtools_app/test/timeline_streams_test.dart:21]: done

**Review by kenzieschmoll — APPROVED** [review]

**Comment by polina-c** [comment]
tested

**Pull request merged**